app/src/main/java/com/example/leagueoflegends/misc/MainAdapter.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.leagueoflegends.misc
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.leagueoflegends.R
+import com.example.leagueoflegends.model.Response
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.champion_main_item.view.*
+
+class MainAdapter: RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+
+    private var champions  = emptyArray<Response.Champion>()
+
+    @JvmName("setChampions1")
+    fun setChampions(champions : Array<Response.Champion>) {
+        this.champions = champions
+        notifyDataSetChanged()
+    }
+
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        val view : View = LayoutInflater.from(parent.context).inflate(R.layout.champion_main_item, parent, false)
+        return ChampionHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        val currentChampion = champions[position]
+        holder.itemView.nameTextView.text = currentChampion.name
+        holder.itemView.titleTextView.text = currentChampion.title
+        holder.itemView.descriptionTextView.text = currentChampion.blurb
+        // Somehow this doesn't work
+        Picasso.get().load("http://ddragon.leagueoflegends.com/cdn/10.21.1/img/champion/${currentChampion.name}.png").into(holder.itemView.avatar)
```

Comment:
the name is not necessarily the same as the id , that might be a problem here

Reply:
also, hardcoded string.
Also, knowing the url is not the responsibility of the view. Put it into a Constants file or something. You might need to change it later, you might need to change just some parts of it later, and you shouldn't need to look through all of your adapters and picture displays to fix 1 URL in the app.